### PR TITLE
Removes unreachable branch on camel-aws-s3 client implementations

### PR DESCRIPTION
**Pull request by orpiske**
I was reviewing the S3 code and came across this code. It seems to be unreachable. 

[x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
[x] Each commit in the pull request should have a meaningful subject line and body.
[x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
[x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
[x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/client/impl/S3ClientIAMOptimizedImpl.java [hunk 1]
```diff
@@ -74,12 +74,14 @@ public AmazonS3 getS3Client() {
 
         if (!configuration.isUseEncryption()) {
             clientBuilder = AmazonS3ClientBuilder.standard().withCredentials(new InstanceProfileCredentialsProvider(false));
-        } else if (configuration.isUseEncryption()) {
+
+            if (configuration.hasProxyConfiguration()) {
+                clientBuilder = clientBuilder.withClientConfiguration(clientConfiguration);
+            }
```
Comment (gnodet) [inline components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/client/impl/S3ClientIAMOptimizedImpl.java:80]: This seems a slightly different issue.  Should this be in a separate commit ?

Comment (orpiske) [reply components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/client/impl/S3ClientIAMOptimizedImpl.java:80]: Absolutely! I can move this change to a separate commit. Let me modify this one and push again.

**Comment by orpiske** [comment]
Adjusting the commit ... an extra file was incorrectly added.

File: components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/client/impl/S3ClientIAMOptimizedImpl.java (hunk 1 above)
Comment (orpiske) [reply components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/client/impl/S3ClientIAMOptimizedImpl.java:80]: I have just moved this part of the PR to a separate commit.

**Comment by orpiske** [comment]
@oscerd @gnodet Anything I can do to help with the review for this PR? If you want, I can describe my reasoning for the change and what, exactly, is the unreachable part. 

Obs.: not trying to be pushy or anything like that. Just trying to learn if I should have done something better on the PR and if I can make your review easier.

**Comment by oscerd** [comment]
I'll review again this when I come back

**Comment by gnodet** [comment]
@oscerd I've merged the commit.  There were 2 branches in a conditional if that could never be executed because they were written like:
```
  if ([condition]) {
   ...
  } else if (![condition]) {
   ...
  } else {
   ...
  }
```
So the last branch can never be executed.

**Comment by oscerd** [comment]
My bad...
